superduperdb/db/base/db.py

```diff
@@ -290,15 +288,31 @@ def delete(self, delete: Delete) -> DeleteResult:
 
         :param delete: delete query object
         """
-        return delete(self)
+        return delete.execute(self)
 
-    def insert(self, insert: Insert) -> InsertResult:
+    def insert(
+        self, insert: Insert, refresh: bool = True, encoders: t.Sequence[Encoder] = ()
+    ) -> InsertResult:
         """
         Insert data.
 
         :param insert: insert query object
         """
-        return insert(self)
+        for e in encoders:
+            self.add(e)
+        for r in insert.documents:
+            r['_fold'] = 'train'  # type: ignore[assignment]
+            if random.random() < s.CFG.fold_probability:
+                r['_fold'] = 'valid'  # type: ignore[assignment]
+        inserted_ids = insert.execute(self)
+        if refresh and s.CFG.cluster.cdc:
+            raise Exception('CFG.cluster.cdc cannot be activated and refresh=True')
+
+        if not s.CFG.cluster.cdc and refresh:
```

Comment:
@blythed please check query.py of mongodb in main branch we are using CDC_COLLECTOIN_LOCKs